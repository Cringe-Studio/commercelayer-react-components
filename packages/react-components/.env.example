# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
