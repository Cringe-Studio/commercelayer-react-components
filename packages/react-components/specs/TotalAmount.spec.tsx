import React from 'react'
import { TotalAmount } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'
import BaseOrderPrice from '../#components/utils/BaseOrderPrice'
import Parent from '../#components/utils/Parent'

const propTypes = components.TotalAmount.propTypes

test('<TotalAmount/>', () => {
  expect.assertions(3)
  const component = renderer.create(<TotalAmount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.format).toBe(propTypes.format)
})

test('<TotalAmount children rendered />', () => {
  expect.assertions(3)
  const component = renderer.create(<TotalAmount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.nodeType).toBe('component')
  expect(rendered.type).toBe(BaseOrderPrice)
})

test('<TotalAmount with custom children />', () => {
  expect.assertions(5)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <TotalAmount format="cents">{CustomComponent}</TotalAmount>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const parentRendered = root.rendered.rendered
  const childRendered = parentRendered.rendered
  expect(tree).toMatchSnapshot()

  expect(rendered.props.children).toBe(CustomComponent)

  expect(parentRendered.nodeType).toBe('component')
  expect(parentRendered.type).toBe(Parent)

  expect(childRendered.props.price).not.toBeUndefined()
})
