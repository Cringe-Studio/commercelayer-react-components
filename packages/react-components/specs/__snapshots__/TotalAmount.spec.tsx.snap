// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalAmount children rendered /> 1`] = `
<span>
  
</span>
`;

exports[`<TotalAmount with custom children /> 1`] = `<span />`;

exports[`<TotalAmount/> 1`] = `
<span>
  
</span>
`;
